R/LD.R

```diff
@@ -182,8 +182,24 @@ extract_LD_for_region <- function(LD_matrix, variants, region, extract_coordinat
 
 # Create a combined LD matrix from multiple matrices
 create_combined_LD_matrix <- function(LD_matrices, variants) {
-    # Create a block matrix with correct names
-    combined_LD_matrix <- as.matrix(do.call(bdiag, LD_matrices))
+    # Extract unique variant names from the list of variants
+    unique_variants <- unique(variants$variants)
```

Comment:
I worry that the unique() function will completely mess up the order of variants. I think the issue you are trying to address here is very specific. We should not use this. We should just address the one variant that joins the two matrices.

Reply:
How about this:

```r
combined_LD_variants <- lapply(LD_matrices_list, function(x) x$extracted_LD_variants)

create_combined_LD_matrix <- function(LD_matrices, variants) {

mergeVectors <- function(vectors) {
  merged <- vectors[[1]]
  sapply(vectors[-1], function(v) { if (tail(merged, 1) == v[1]) merged <<- c(merged, v[-1]) else merged <<- c(merged, v) })
  merged
}
    
unique_variants <- mergeVectors(variants)  

```